kubemark tests are failing

### Which jobs are failing?

All kubemark tests started failing around 11-04.

* https://k8s-testgrid.appspot.com/sig-scalability-gce
* https://k8s-testgrid.appspot.com/sig-scalability-kubemark#kubemark-master-500

The testing framework is not able to schedule pods. hollow kubelets has errors like:

https://storage.googleapis.com/sig-scalability-logs/ci-kubernetes-kubemark-100-gce/1588607954326654976/kubemark-100-minion-group-d6kl/kubelet-hollow-node-sjkkm.log 

```
E1104 19:24:13.589072      11 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 138 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x3d7a4a0?, 0x710eb30})
        vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000226600?})
        vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x3d7a4a0, 0x710eb30})
        /usr/local/go/src/runtime/panic.go:884 +0x212
k8s.io/kubernetes/pkg/kubelet/config.(*podStorage).merge.func1({0xc0001242e0?, 0x46dd039?, 0x3?}, 0x3?, 0xc000aa0a70?)
        pkg/kubelet/config/config.go:248 +0x19a
k8s.io/kubernetes/pkg/kubelet/config.(*podStorage).merge(0xc000226600, {0x46dd039, 0x3}, {0x4121080?, 0xc0008d2960})
        pkg/kubelet/config/config.go:298 +0x417
k8s.io/kubernetes/pkg/kubelet/config.(*podStorage).Merge(0xc000226600, {0x46dd039, 0x3}, {0x4121080, 0xc0008d2960})
        pkg/kubelet/config/config.go:166 +0x126
k8s.io/kubernetes/pkg/util/config.(*Mux).listen(...)
        pkg/util/config/config.go:85
k8s.io/kubernetes/pkg/util/config.(*Mux).ChannelWithContext.func1()
        pkg/util/config/config.go:79 +0x71
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00006dfa0?)
        vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x1?, {0x4e8aac0, 0xc00004ec60}, 0x1, 0x0)
        vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x16c749a?, 0x0, 0x0, 0x60?, 0x443d05?)
        vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0xc000214360?, 0x0?, 0xc000214360?)
        vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:92 +0x25
created by k8s.io/kubernetes/pkg/util/config.(*Mux).ChannelWithContext
        pkg/util/config/config.go:79 +0x245
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
```

Finally (when the test already failed) it changes to
```
I1104 19:29:25.700264      11 kubelet_node_status.go:70] "Attempting to register node" node="hollow-node-sjkkm"
E1104 19:29:25.800157      11 kubelet.go:2455] "Error getting node" err="node \"hollow-node-sjkkm\" not found"
E1104 19:29:25.900640      11 kubelet.go:2455] "Error getting node" err="node \"hollow-node-sjkkm\" not found"
E1104 19:29:26.001376      11 kubelet.go:2455] "Error getting node" err="node \"hollow-node-sjkkm\" not found"
E1104 19:29:26.102122      11 kubelet.go:2455] "Error getting node" err="node \"hollow-node-sjkkm\" not found"
E1104 19:29:26.202734      11 kubelet.go:2455] "Error getting node" err="node \"hollow-node-sjkkm\" not found"
```

Suspected commit range: https://github.com/kubernetes/kubernetes/compare/c8a3657bd...63a62defd

### Which tests are failing?

* ci-kubernetes-kubemark-500-gce.Overall

### Since when has it been failing?

11-04

### Testgrid link

https://k8s-testgrid.appspot.com/sig-scalability-kubemark#kubemark-100

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig scalability

--- a/pkg/kubemark/hollow_kubelet.go
+++ b/pkg/kubemark/hollow_kubelet.go
@@ -36,6 +36,7 @@ import (
 	containertest "k8s.io/kubernetes/pkg/kubelet/container/testing"
 	probetest "k8s.io/kubernetes/pkg/kubelet/prober/testing"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	kubeletutil "k8s.io/kubernetes/pkg/kubelet/util"
 	"k8s.io/kubernetes/pkg/util/oom"
 	"k8s.io/kubernetes/pkg/volume"
 	"k8s.io/kubernetes/pkg/volume/cephfs"
@@ -94,21 +95,22 @@ func NewHollowKubelet(
 	runtimeService internalapi.RuntimeService,
 	containerManager cm.ContainerManager) *HollowKubelet {
 	d := &kubelet.Dependencies{
-		KubeClient:           client,
-		HeartbeatClient:      heartbeatClient,
-		ProbeManager:         probetest.FakeManager{},
-		RemoteRuntimeService: runtimeService,
-		RemoteImageService:   imageService,
-		CAdvisorInterface:    cadvisorInterface,
-		Cloud:                nil,
-		OSInterface:          &containertest.FakeOS{},
-		ContainerManager:     containerManager,
-		VolumePlugins:        volumePlugins(),
-		TLSOptions:           nil,
-		OOMAdjuster:          oom.NewFakeOOMAdjuster(),
-		Mounter:              &mount.FakeMounter{},
-		Subpather:            &subpath.FakeSubpath{},
-		HostUtil:             hostutil.NewFakeHostUtil(nil),
+		KubeClient:               client,
+		HeartbeatClient:          heartbeatClient,
+		ProbeManager:             probetest.FakeManager{},
+		RemoteRuntimeService:     runtimeService,
+		RemoteImageService:       imageService,
+		CAdvisorInterface:        cadvisorInterface,
+		Cloud:                    nil,
+		OSInterface:              &containertest.FakeOS{},
+		ContainerManager:         containerManager,
+		VolumePlugins:            volumePlugins(),
+		TLSOptions:               nil,
+		OOMAdjuster:              oom.NewFakeOOMAdjuster(),
+		Mounter:                  &mount.FakeMounter{},
+		Subpather:                &subpath.FakeSubpath{},
+		HostUtil:                 hostutil.NewFakeHostUtil(nil),
+		PodStartupLatencyTracker: kubeletutil.NewPodStartupLatencyTracker(),
 	}
 
 	return &HollowKubelet{

